cairo/ethereum/cancun/vm.cairo

```diff
@@ -88,6 +105,313 @@ struct MessageStruct {
     parent_evm: OptionalEvm,
 }
 
+func incorporate_child_on_success{range_check_ptr, poseidon_ptr: PoseidonBuiltin*, evm: Evm}(
+    child_evm: Evm
+) {
+    alloc_locals;
+
+    // TODO: unless we want to retro-prove all blocks, we could remove this logic.
+    // In block 2675119, the empty account at 0x3 (the RIPEMD160 precompile) was
+    // cleared despite running out of gas. This is an obscure edge case that can
+    // only happen to a precompile.
+    // According to the general rules governing clearing of empty accounts, the
+    // touch should have been reverted. Due to client bugs, this event went
+    // unnoticed and 0x3 has been exempted from the rule that touches are
+    // reverted in order to preserve this historical behaviour.
```

Comment:
This comment should be in `incorporate_child_on_error`
A link to the EELS would be relevant also